### PR TITLE
[PM-9755] Change error message from a toast to a dialog

### PR DESCRIPTION
**Pull request by andrebispo5**
## 🎟️ Tracking

[PM-9755](https://bitwarden.atlassian.net/browse/PM-9755)

## 📔 Objective

On some devices, the error message for the null Duo AuthUrl did not display correctly in the toast, so it was converted to an error dialog.

## 📸 Screenshots
image

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-9755]: https://bitwarden.atlassian.net/browse/PM-9755?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginViewModel.kt [hunk 1]
```diff
@@ -205,11 +205,17 @@ class TwoFactorLoginViewModel @Inject constructor(
...
                 // The url should not be empty unless the environment is somehow not supported.
-                sendEvent(
-                    event = authUrl
-                        ?.let { TwoFactorLoginEvent.NavigateToDuo(uri = Uri.parse(it)) }
-                        ?: TwoFactorLoginEvent.ShowToast(R.string.error_connecting_with_the_duo_service_use_a_different_two_step_login_method_or_contact_duo_for_assistance.asText()),
-                )
+                authUrl
+                    ?.let {
+                    TwoFactorLoginEvent.NavigateToDuo(uri = Uri.parse(it))
+                } ?: mutableStateFlow.update {
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginViewModel.kt:211]: The chaining should have the elvis on it's own line. Also, looks like the indentation was not updated here.

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginViewModel.kt [hunk 2]
```diff
@@ -205,11 +205,17 @@ class TwoFactorLoginViewModel @Inject constructor(
...
                 val authUrl = authRepository.twoFactorResponse.twoFactorDuoAuthUrl
                 // The url should not be empty unless the environment is somehow not supported.
-                sendEvent(
-                    event = authUrl
-                        ?.let { TwoFactorLoginEvent.NavigateToDuo(uri = Uri.parse(it)) }
-                        ?: TwoFactorLoginEvent.ShowToast(R.string.error_connecting_with_the_duo_service_use_a_different_two_step_login_method_or_contact_duo_for_assistance.asText()),
-                )
+                authUrl
+                    ?.let {
+                    TwoFactorLoginEvent.NavigateToDuo(uri = Uri.parse(it))
```
[the remark this answers is not in the document]
Comment (andrebispo5) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginViewModel.kt:210]: Removed it in one of the refactor iteration and forgot about it ! Thanks

[the remark this answers is not in the document]
Comment (andrebispo5) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginViewModel.kt]: Left the `sendEvent` in the same line so it won't bother with the long method lint error.

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginViewModel.kt [hunk 3]
```diff
@@ -205,11 +205,18 @@ class TwoFactorLoginViewModel @Inject constructor(
             -> {
                 val authUrl = authRepository.twoFactorResponse.twoFactorDuoAuthUrl
                 // The url should not be empty unless the environment is somehow not supported.
-                sendEvent(
-                    event = authUrl
-                        ?.let { TwoFactorLoginEvent.NavigateToDuo(uri = Uri.parse(it)) }
-                        ?: TwoFactorLoginEvent.ShowToast(R.string.error_connecting_with_the_duo_service_use_a_different_two_step_login_method_or_contact_duo_for_assistance.asText()),
-                )
+                authUrl
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginViewModel.kt:208]: 👍

**Pull request merged**